content/docs/thinking-in-react.md

```diff
@@ -1,24 +1,24 @@
 ---
 id: thinking-in-react
-title: Thinking in React
+title: React में सोचना 
 permalink: docs/thinking-in-react.html
 redirect_from:
   - 'blog/2013/11/05/thinking-in-react.html'
   - 'docs/thinking-in-react-zh-CN.html'
 prev: composition-vs-inheritance.html
 ---
 
-React is, in our opinion, the premier way to build big, fast Web apps with JavaScript. It has scaled very well for us at Facebook and Instagram.
+हमारी राय में, जावास्क्रिप्ट द्वारा, बड़ी और तेज़ वेब एप्पस बनाने का React एक मुख्य तरीका है। यह हमारे लिए Facebook और Instagram के लिए काफी अच्छे से बढ़ा है।
 
-One of the many great parts of React is how it makes you think about apps as you build them. In this document, we'll walk you through the thought process of building a searchable product data table using React.
+कई में से एक मुख्य बात React की है कि कैसे यह आपको एप्प बनाते हुए उसके बारे में सोचने पर मजबूर करता है। इस डॉक्यूमेंट में हम React से बने सर्चएब्ल प्रोडक्ट डाटा टेबल को बनाने की विचार प्रक्रिया पर ध्यान देंगे।
 
-## Start With A Mock {#start-with-a-mock}
+## मॉक के साथ शुरुवात {#start-with-a-mock}
 
-Imagine that we already have a JSON API and a mock from our designer. The mock looks like this:
+कल्पना कीजिये कि हमारे पास पहले से JSON API और हमारे डिज़ाइनर से एक मॉक उपलब्ध है। मॉक ऐसा कुछ दिखता है:
 
 ![Mockup](../images/blog/thinking-in-react-mock.png)
 
-Our JSON API returns some data that looks like this:
+हमारी JSON API कुछ ऐसा डाटा वापस भेजेगी:
 
```

Comment:
words in code such as sporting goods and names could be translated as well

Reply:
Didn't quite get you... Please elaborate where are you referring to?

Reply:
the keys won't be translated, only values. Sporting Goods, electronics, Football etc.